server/test/filters/UnsupportedBrowserFilterTest.java

```diff
@@ -0,0 +1,90 @@
+package filters;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.test.Helpers.fakeRequest;
+
+import akka.stream.testkit.NoMaterializer$;
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import play.libs.streams.Accumulator;
+import play.mvc.EssentialAction;
+import play.mvc.Http;
+import play.mvc.Result;
+import play.mvc.Results;
+
+public class UnsupportedBrowserFilterTest {
+
+  private final UnsupportedBrowserFilter filter = new UnsupportedBrowserFilter();
+  private final String unsupportedBrowserPath =
+      controllers.routes.SupportController.handleUnsupportedBrowser().path();
+
+  private static final String IE11 =
+      "Mozilla/5.0 (Windows NT 10.0; Trident/7.0; rv:11.0) like Gecko";
+  private static final String IE10 =
+      "Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; Trident/6.0)";
+  private static final String IE9 =
+      "Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)";
+  private static final String EDGE =
+      "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko)"
+          + " Chrome/103.0.0.0 Safari/537.36 Edg/103.0.1264.49";
+  private static final String FIREFOX =
+      "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:102.0) Gecko/20100101 Firefox/102.0";
+  private static final String CHROME =
+      "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko)"
+          + " Chrome/103.0.0.0 Safari/537.36";
+  private static final String SAFARI =
+      "Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/605.1.15 (KHTML, like Gecko)"
+          + " Version/15.4 Safari/605.1.15";
+
+  @Test
+  public void testMissingUserAgentDoesNotTriggerRedirect() throws Exception {
+    Result res = runFilter(fakeRequest(), Results.ok(""));
+    assertThat(res.status()).isEqualTo(200);
+  }
+
+  @Test
+  public void testOldIETriggerRedirect() throws Exception {
```

Comment:
Not blocking: Nit. You can use a parameterized test here so that you'll get individual test names for each of the things being iterated (https://github.com/seattle-uat/civiform/blob/7fea10a885d891f8926bf1fbd24ca0148f9e7732/server/test/repository/ProgramRepositoryTest.java#L234).

A guide to the general JUnit mechanism: https://www.baeldung.com/parameterized-tests-junit-5

Same feedback applies to the other browser tests.

Reply:
I don't think I can use it here unfortunately. We are using Junit4 for unit tests `org.junit.*` packages. Juni4 allows you to parameterize whole test class only, it doesn't allow to parameterize individual test methods.

The link you are mentioned is junit5 which uses `org.junit.jupiter.*` packages. If I were to use it - we should probably migrate all existing test to junit5 and mandate usage of junit5. And that requires some SBT [plugin](https://github.com/sbt/sbt-jupiter-interface) it seems.

Reply:
Oh, we are actually using some third-party plugin that you referenced: JUnitParamsRunner. So it's not junit5 thing. Ok, I'll rewrite the test then.

Reply:
Apologies for the poor link :(